pulp_container/tests/functional/api/test_push_content.py

```diff
@@ -36,32 +29,14 @@
 class PushContentTestCase(unittest.TestCase):
     """Verify whether images can be pushed to pulp."""
 
-    def setUp(self):
-        """Create class-wide variables.
-
-        1. Create a repository.
-        2. Create a container distribution to serve the repository
-        """
-        # Step 1
-        self.repo = repositories_api.create(ContainerContainerRepository(**gen_repo()))
-        # cls.teardown_cleanups.append((cls.repositories_api.delete, _repo.pulp_href))
-        self.addCleanup(repositories_api.delete, self.repo.pulp_href)
-
-        # Step 2
-        distribution_response = distributions_api.create(
-            ContainerContainerDistribution(
-                **gen_distribution(repository=self.repo.pulp_href)
-            )
-        )
-        created_resources = monitor_task(distribution_response.task)
-        self.distribution = distributions_api.read(created_resources[0])
-        self.addCleanup(distributions_api.delete, self.distribution.pulp_href)
-
     def test_push_using_podman(self):
         """Test push with official registry client"""
         # TODO better handling of the "http://"
-        local_url = urljoin(cfg.get_base_url(), self.distribution.base_path)[7:]
+        local_url = urljoin(cfg.get_base_url(), 'foo/bar:1.0')[7:]
         registry.pull("busybox:latest")
         registry.tag("busybox:latest", local_url)
         registry.push(local_url)
-        self.assertTrue(True)
+        repository = repositories_api.list(name='foo/bar').results[0]
+        distribution = distributions_api.list(name='foo/bar').results[0]
+        self.addCleanup(repositories_api.delete, repository.pulp_href)
```

Comment:
why do we cleanup those resources? subsequent re-pushes should work  if repo/distribution  is already present

Reply:
Tests usually clean up anything they create. This test creates a repository and distribution.